bindings/go/oci/internal/lister/component/component_versions.go

```diff
@@ -123,3 +130,17 @@ func ReferenceTagVersionResolver(component string, store interface {
 	}
 	return tagResolver
 }
+
+// validateOldOCMComponentVersion checks for old component versions pre-2024 which didn't have an annotation.
+// In that case, we check if the manifest has a config of type ocm config and if yes, we immediately just return
+// the tag as valid. We only do this for manifest layers and not index layers because pre-2024 component versions
+// didn't have indexes.
+func validateOldOCMComponentVersion(manifest ociImageSpecV1.Manifest) bool {
+	// if validation is present, we have a new ( older than 2024 October ) component version
+	_, ok := manifest.Annotations[annotations.OCMComponentVersion]
```

Comment:
I think now the validation you have is run twice, once in line 111 and once here.

Reply:
Okay, this is nicer. :)